signer/PvGuard.go

```diff
@@ -8,30 +8,35 @@ import (
 	tm "github.com/tendermint/tendermint/types"
 )
 
+var _ PrivValidator = &PvGuard{}
+
 // PvGuard guards access to an underlying PrivValidator by using mutexes
 // for each of the PrivValidator interface functions
 type PvGuard struct {
```

Comment:
`SyncPrivValidator` would probably be a bit more idiomatic and clearer name, granting synchronized access to a non-goroutine-safe type. But that can change in another PR, not in scope for this one.

Reply:
I already renamed it to SingleSignerValidator in the ICS PR, agreed PvGuard is unclear.